internal/driver/gomobile/file_android.go

```diff
@@ -12,18 +12,34 @@ char* readStream(uintptr_t jni_env, uintptr_t ctx, void* stream, int len, int* t
 void* saveStream(uintptr_t jni_env, uintptr_t ctx, char* uriCstr);
 void writeStream(uintptr_t jni_env, uintptr_t ctx, void* stream, char* data, int len);
 void closeStream(uintptr_t jni_env, uintptr_t ctx, void* stream);
+char* contentURIGetFileName(uintptr_t jni_env, uintptr_t ctx, char* uriCstr);
 */
 import "C"
 import (
 	"errors"
 	"io"
 	"os"
+	"path/filepath"
 	"unsafe"
 
 	"fyne.io/fyne/v2/storage/repository"
 	"github.com/fyne-io/mobile/app"
 )
 
+func (m mobileURI) Name() string {
```

Comment:
We try to keep things ordered alphabetically (type, then method) - more info at https://github.com/fyne-io/fyne/wiki/Contributing.
So I think mobileURI should come after javaStream.
I wondered also if there was a reason for not using a pointer receiver here, when you create the type as a pointer?

Reply:
Or (from note above) perhaps it goes at the end of `repository.go`?
It depends on whether iOS should use this functionality or not I think.